### DISCUSSION TITLE
Tensor coefficient Poisson equation

### DISCUSSION THREAD

**Discussion by kaigaox** [mfem/mfem#2086] [Q&A] [answered]
Hi, does anyone konw how to use MFEM to solve a tensor coefficient Poisson equation in the form of
![image](https://user-images.githubusercontent.com/10822134/110169525-0c5eee80-7db6-11eb-9b66-925f07e30abb.png)
where c is a 3x3 tensor in the form of 
![image](https://user-images.githubusercontent.com/10822134/110170042-d2421c80-7db6-11eb-904b-f467347d16e5.png)

I want to solve the Poisson in such a way that, I have a cube domain that has n1xn2xn3 elements, and for each element, I want to specify some value for the c tensor. 

Thanks a lot

**Comment by jandrej** [comment]
Hello,

does one of the [MatrixCoefficients](https://mfem.github.io/doxygen/html/classmfem_1_1MatrixCoefficient.html#a40686042c723f31c955a56705fb6c525) work for you?

**Comment by v-dobrev** [comment, marked as answer]
You can define a custom `MatrixCoefficicient`, e.g. `ElementwiseMatrixCoefficient`, to use with the `DiffusionIntegrator` in a way very similar to Example 1.

For the `ElementwiseMatrixCoefficient` you just need to implement the virtual method `Eval(DenseMatrix &K, ElementTransformation &T, const IntegrationPoint &ip)` -- you need to set the output parameter `K` based on the input element index which is given by `T.ElementNo`.
